js/src/api/contract/contract.js

```diff
@@ -133,9 +133,11 @@ export default class Contract {
 
         return this._api.parity
           .postTransaction(encodedOptions)
-          .then((requestId) => {
-            statecb(null, { state: 'checkRequest', requestId });
-            return this._pollCheckRequest(requestId);
+          .then((result) => {
+            if (result.length !== 66) {
+              statecb(null, { state: 'checkRequest', result });
```

Comment:
Why did we lose the `requestId` naming? This is is that the holder of the statecb checks the `requestId` field which does not exist anywhere anymore.

Reply:
oh yeah that's my mistake, one sec I will revise

Reply:
revised 9deac99a718fa351a28ab5b254928ecac9e4522b